src/vos/vos_internal.h

```diff
@@ -1183,6 +1184,25 @@ vos_epc_punched(daos_epoch_t epc, uint16_t minor_epc,
 	return false;
 }
 
+static inline bool
+vos_dtx_hit_inprogress(void)
+{
+	struct dtx_handle	*dth = vos_dth_get();
+
+	return dth != NULL && dth->dth_share_tbd_count > 0;
+}
+
+static inline bool
+vos_dtx_continue_detect(int rc)
+{
+	struct dtx_handle	*dth = vos_dth_get();
+
+	/* Continue to detect other potential in-prepared DTX. */
+	return rc == -DER_INPROGRESS && dth != NULL &&
+		dth->dth_share_tbd_count > 0 &&
+		dth->dth_share_tbd_count < DTX_DETECT_MAX;
+}
+
```

Comment:
the line below is why I don't like the term uncertain for this case.

Reply:
Yes, I noticed that there is already some concept has used the term of "uncertain".